docs/api.md

```diff
@@ -0,0 +1,287 @@
+
+There are three main steps to uploading a file and its metadata to Witan via the API.
+
+### Step 1 - Generate your receipt-id.
+
+Assuming that you have logged in and have an auth token the first step is to do a POST to the `/api/file/upload` endpoint. 
+
+```
+curl -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' --header 'authorization: eyJhb.....7SeMQ' 'https://api.witanforcities.com/api/files/upload'
+```
+
+This will return a receipt-id, using this id you can now poll against the API to get the file upload link.
+
+```
+curl -X GET --header 'Accept: application/json' --header 'authorization: eyJhb.....7SeMQ' 'https://api.witanforcities.com/api/receipts/abcdefg-abcd-abcd-abcd-abcdeghij'
+```
+
+### Step 2 - Get the upload link.
+
+The response from the `/api/receipt/[receipt-id]` call will respond with either a 202 code, meaning the process is still being handled by Witan, you should continue to call the receipt endpoint with the same URL to see if the receipt has updated. 
+
+If you receive a response with a 200 status code then the request is classed as complete and you will see the following response from the API. 
+
+```
+{
+  "witan.httpapi.spec/uri": "https://prod-witan-kixi-datastore-filestore...........",
+  "kixi.datastore.filestore/id": "d1bec41e.....46c2"
+}
+```
+
+It is your responsibility to upload the file you want stored on Witan using the URL that is in the `witan.httpapi.spec/url` response field. 
+
+```
+curl -vvv -XPUT --data-binary /Users/jasonbell/Downloads/resp_gas.xml "https://prod-witan-kixi-datastore-filestore.s3-eu-west-1.amazonaws.com/d1bec41e-086e-4ad7-
+```
+
+### Step 3 - Post the metadata about that file.
+
+Now that the file is uploaded to Witan you are required to send metadata about that file. If you do not do this then your file entry will not show in Witan searches or file lists. 
+
+The minimum information for file metadata is:
+
+```
+{"kixi.datastore.metadatastore/header":false, 
+ "kixi.datastore.metadatastore/size-bytes":13856,
+ "kixi.datastore.metadatastore/name":"my_uploaded_file.xml",
+ "kixi.datastore.metadatastore/file-type":"XML"}
+```
+
+#### Programatically finding the size of your file.
+
+You are required to know the size of your file in bytes, for example if you were using the statistical language R for example you would use the `file.size()` command:
+
+```
+> file.size("/path/to/uploaded/file/your_file.xml")
+[1] 13856
+``` 
+In Python:
+
+```
+>>> import os
+>>> filesize = os.path.getsize("/path/to/uploaded/file/your_file.xml")
+>>> filesize
+13856
+```
+
+In PHP:
+
+```
+<?PHP
+$filename = "/path/to/uploaded/file/your_file.xml";
+var $filesize = filesize($filename);
+?>
+```
+
+In Java:
+
+```
+File file = new File("/path/to/uploaded/file/your_file.xml");
+double filesize = file.length();
```

Comment:
In Bash maybe?

Reply:
Not sure how many would use it but I've added it.